src/server/routes/consentToken.ts

```diff
@@ -26,7 +26,18 @@ router.get(
 
 			trackMetric('ConsentToken::Success');
 
-			return res.redirect(303, buildUrl('/subscribe/success'));
+			return res.redirect(
+				303,
+				buildUrlWithQueryParams(
+					'/subscribe/success',
+					{},
+					{
+						ref: req.query.ref?.toString(),
+						refViewId: req.query.refViewId?.toString(),
+						returnUrl: '',
+					},
```

Comment:
rather than calling `req.query` directly here, use `res.locals.queryParams`

```suggestion
					res.locals.queryParams,
```

Docs: https://github.com/guardian/gateway/blob/main/docs/development.md#query-params

And info on how to use them server side: https://github.com/guardian/gateway/blob/main/docs/development.md#server

the tl;dr is that we have middleware which parses and checks all query params for us making sure they're all valid, and we make them accessible through the request state `res.locals`, this includes the `ref` and `refViewId` params